From 9f878e35c6f635732d0922442f61d6a3aeb43270 Mon Sep 17 00:00:00 2001
From: Alan Winkowski <walan@marvell.com>
Date: Wed, 6 Sep 2017 18:19:34 +0300
Subject: [PATCH 1232/1345] fix: net: mvpp2x: skip disable of tx interrupts in
 MUSDK mode

commit  06d7c1f29901a0eeb5092d8699a74771bf6c866f from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- This patch fixes an issue when a MUSDK port modifies the
  platform_device wide interrupt_tx_done attribute.
  It should not in any case.
- This patch does not affect normal mvpp2x behavior (w/o musdk patches).
- fixes A7K8K-3242

Change-Id: I9299ab974a8c265902dd28b6f65eb3261c98d806
Signed-off-by: Alan Winkowski <walan@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43857
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 8f89ab7..8b541e1 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5157,7 +5157,8 @@ static int mv_pp2x_port_probe(struct platform_device *pdev,
 	mv_pp2x_check_queue_size_valid(port);
 
 	if (mv_pp2_num_cpu_irqs(port) < num_active_cpus() &&
-	    port->priv->pp2xdata->interrupt_tx_done) {
+	    port->priv->pp2xdata->interrupt_tx_done &&
+	    (!(port->flags & (MVPP2_F_IF_MUSDK | MVPP2_F_LOOPBACK)))) {
 		port->priv->pp2xdata->interrupt_tx_done = false;
 		dev_info(&pdev->dev, "mvpp2x: interrupt_tx_done override to false\n");
 	}
-- 
1.7.9.5

